kyo-prelude/shared/src/main/scala/kyo2/kernel/Safepoint.scala

```diff
@@ -120,25 +121,32 @@ object Safepoint:
                 finally self.exit(depth)
     end handle
 
-    private def handle(self: Safepoint, cause: Throwable): Nothing =
-        val size  = Math.min(self.traceIdx, maxTraceFrames)
-        val trace = copyTrace(self.trace, self.traceIdx).filter(_ != null).map(_.parse)
-        val toPad = trace.map(_.snippetShort.size).max + 1
-        val elements = trace.map { frame =>
-            StackTraceElement(
-                frame.snippetShort.reverse.padTo(toPad, ' ').reverse + " @ " + frame.declaringClass,
-                frame.methodName,
-                frame.position.fileName,
-                frame.position.lineNumber
+    private[kyo2] def insertTrace(cause: Throwable)(using self: Safepoint): Unit =
+        val size = Math.min(self.traceIdx, maxTraceFrames)
+        if size > 0 then
+            val trace = copyTrace(self.trace, self.traceIdx).filter(_ != null).map(_.parse)
+            val toPad = trace.map(_.snippetShort.size).maxOption.getOrElse(0) + 1
+            val elements =
+                trace.foldLeft(List.empty[Frame.Parsed]) {
+                    case (acc, curr) =>
+                        acc match
+                            case `curr` :: tail => acc
+                            case _              => curr :: acc
+                }.reverse.map { frame =>
+                    StackTraceElement(
+                        frame.snippetShort.reverse.padTo(toPad, ' ').reverse + " @ " + frame.declaringClass,
+                        frame.methodName,
+                        frame.position.fileName,
+                        frame.position.lineNumber
+                    )
+                }.reverse
+            val prefix = cause.getStackTrace.takeWhile(e =>
+                e.getFileName() != elements(0).getFileName() || e.getLineNumber != elements(0).getLineNumber()
```

Comment:
I noticed this edge case with fibers: when a fiber starts executing and fails before reaching a transformation that pushes a frame, `elements` is empty. I've added a guard `size > 0` to fix it.